### PR TITLE
CAS-8919 modify getRowStats() to use _subScanProps if it is cached

### PR DESCRIPTION
**Pull request by dmehring**
Also add correct spelling of getUniqueFieldIDs() and deprecate getUniqueFiedIDs().

### CODE REVIEW

File: ms/MSOper/MSMetaData.h [hunk 1]
```diff
@@ -85,12 +85,13 @@ class MSMetaData {
         // the subscan and that spwID
         map<uInt, Quantity> meanInterval;
         Quantity meanExposureTime;
-        uInt nrows;
         std::set<uInt> spws;
         // number of rows for each spectral window
         map<uInt, uInt> spwNRows;
         std::set<Int> stateIDs;
-        std::map<Double, TimeStampProperties> timeProps;
+        map<Double, TimeStampProperties> timeProps;
```
Comment (juliantaylor) [inline ms/MSOper/MSMetaData.h]: it needs to be std::map, that you can use it without is an error, std namespace should not be set default in headers

File: ms/MSOper/MSMetaData.h [hunk 2]
```diff
@@ -85,12 +85,13 @@ class MSMetaData {
...
+        // number of cross-correlation rows.
+        uInt xcRows;
```
Comment (juliantaylor) [inline ms/MSOper/MSMetaData.h]: this should maybe be placed next to acRows as it is related
as you are breaking binary compat anyway be removing nrows you don't need to add at the bottom.

File: ms/MSOper/MSMetaData.h (hunk 1 above)
Comment (dmehring) [inline ms/MSOper/MSMetaData.h]: This is the first I'm hearing about this. Why is it not permitted to use using namespace std; ?

Comment (juliantaylor) [inline ms/MSOper/MSMetaData.h]: because this is a header, using namespace cannot be reverted so you are forcing it on everybody including the header.

Comment (juliantaylor) [inline ms/MSOper/MSMetaData.h]: the actual culprit is in some other header, but that is no reason to stick to bad style, also it avoids an error in for the case when the cause is fixed.

Comment (dmehring) [inline ms/MSOper/MSMetaData.h]: OK, fair enough. Thanks.

**Comment by dmehring** [comment]
moved xcRows as requested. Changed all map to std::map in header file.

**Comment by dmehring** [comment]
Thanks!

**Pull request merged**